src/base/serial/tty_io.c

```diff
@@ -299,22 +300,22 @@ static int tty_already_locked(char *nam)
  */
 static int tty_lock(const char *path, int mode)
 {
-  char saved_path[strlen(config.tty_lockdir) + 1 +
-                  strlen(config.tty_lockfile) +
-                  strlen(path) + 1];
+  char saved_path[PATH_MAX];
   struct passwd *pw;
   pid_t ime;
   const char *slash;
 
-  if (path == NULL) return(0);        /* standard input */
+  if (path == NULL)       /* standard input */
+    return 0;
+
   slash = strrchr(path, '/');
   if (slash == NULL)
     slash = path;
   else
     slash++;
 
-  sprintf(saved_path, "%s/%s%s", config.tty_lockdir, config.tty_lockfile,
-	  slash);
+  snprintf(saved_path, PATH_MAX - 1, "%s/%s%s",
```

Comment:
No need for `-1`.
You can as well apply `sizeof()`.

Reply:
Yes, embarrassingly showing my recent lack of coding practice!